core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -171,6 +171,23 @@
   private volatile MapIterator destructiveIterator = null;
   private LinkedList<UnsafeSorterSpillWriter> spillWriters = new LinkedList<>();
 
+  private boolean anyNullKeyExists = false;
+
+  public boolean inputEmpty()
+  {
+    return ((numKeys == 0) && !anyNullKeyExists);
```

Comment:
nit: The outer params can be dropped ?

Reply:
> I am still a bit surprised that you didn't have to modify any .sql.out files because the plan would have changed from BNLJ to BHJ.

.sql.out only verified the output schema and output answer, as long as the change itself could come out with right answer, the verified will pass.

Reply:
done